mozci/console/commands/push.py

```diff
@@ -124,6 +134,40 @@ def classify_commands_pushes(
     return pushes
 
 
+def retrieve_classify_parameters(medium_conf_option, high_conf_option):
+    classify_parameters = {}
+
+    for parameter_name in CLASSIFY_PARAMETERS:
+        if config.get(parameter_name) is not None:
+            classify_parameters[parameter_name] = config[parameter_name]
+
+    parameter_name = "intermittent_confidence_threshold"
+    if medium_conf_option is not None:
+        if classify_parameters.get(parameter_name) is not None:
+            raise Exception(
+                f"You should provide either --medium-confidence CLI option OR '{parameter_name}' in the config secret not both."
+            )
+
+        try:
+            classify_parameters[parameter_name] = float(medium_conf_option)
+        except ValueError:
+            raise Exception("Provided --medium-confidence should be a float.")
+
+    parameter_name = "real_confidence_threshold"
+    if high_conf_option is not None:
+        if classify_parameters.get(parameter_name) is not None:
+            raise Exception(
+                f"You should provide either --high-confidence CLI option OR '{parameter_name}' in the config secret not both."
+            )
+
+        try:
+            classify_parameters[parameter_name] = float(high_conf_option)
+        except ValueError:
+            raise Exception("Provided --high-confidence should be a float.")
```

Comment:
Could we have a list of options through `Command` so we don't have to check them one by one manually?

Reply:
If it isn't feasible, we could define options for all of the values in CLASSIFY_PARAMETERS, and then use CLASSIFY_PARAMETERS as a list (checking that no items in CLASSIFY_PARAMETERS are both in config and passed as parameters)

Reply:
I was asking to avoid hard-coding the option names here, but check if it is possible to get all option names from the `Command` instance.
So you'd do something like:
```
    for parameter_name in self.option_names():
        if config.get(parameter_name) is not None and self.options(parameter_name) is not None:
            raise Exception(
                f"You should provide either --{parameter_name} CLI option OR '{parameter_name}' in the config secret not both."
            )
```

If that isn't feasible, then do something like:
```
    for parameter_name in CLASSIFY_PARAMETERS:
        if config.get(parameter_name) is not None and self.options(parameter_name) is not None:
            raise Exception(
                f"You should provide either --{parameter_name} CLI option OR '{parameter_name}' in the config secret not both."
            )
```
possibly defining all CLASSIFY_PARAMETERS as options (this last point is not required as we are not using them right now like you said, but nice to have as we might want to use them in the future for local runs of classify eval, if we want to run multiple in parallel and so using config isn't feasible).

Reply:
They'd just be ignored as they are not also configuration options

Reply:
I pushed a new commit where I rely on the `classify` function signature instead of a constant to get the supported parameters. I also added support for all parameters as CLI options.

For each parameter, one can :
- define nothing, then the default value specified in the `classify` signature will be used.
- set the parameter via config **AND** via a CLI option, then an error will be raised.
- define the parameter via the config, it will be checked that the provided value is castable to the actual type of the parameter.
- define the parameter via CLI option, it will be checked that the provided value is castable to the actual type of the parameter.